Sketch.java

```diff
@@ -1,36 +1,70 @@
 import processing.core.PApplet;
+import processing.core.PImage;
 
 public class Sketch extends PApplet {
-	
-	
-  /**
-   * Called once at the beginning of execution, put your size all in this method
-   */
+
+  // variables
+  PImage imgSetting;
+  PImage imgAngryMan;
+  float fltAngryManX = 0;
+	float fltAngryManY = 0;
+  // The reason this movement is not considered linear movement is because there is an exponent which means it's non linear movement. It is quadratic, y=x^2
+	double fltAngryManSpeedX = 10;
+  double fltAngryManSpeedY = Math.pow(fltAngryManSpeedX,2);
+  
+  float fltCircleX = 0;
+	float fltCircleY = 0;
+  double fltCircleSpeedX = 5;
+  double fltCircleSpeedY = Math.pow(fltCircleSpeedX,2);
+
+  // set size
   public void settings() {
-	// put your size call here
-    size(400, 400);
+    size(700, 700);
   }
 
-  /** 
-   * Called once at the beginning of execution.  Add initial set up
-   * values here i.e background, stroke, fill etc.
-   */
+
   public void setup() {
-    background(210, 255, 173);
+    //background colour
+    background(0);
+
+    //load and resize images
+    imgSetting = loadImage("Setting.jpg");
+    imgSetting.resize(width, height);
+
+    imgAngryMan = loadImage("AngryMan.jpg");
+    imgAngryMan.resize(200, 200);
   }
 
-  /**
-   * Called repeatedly, anything drawn to the screen goes here
-   */
+
   public void draw() {
-	  
-	// sample code, delete this stuff
-    stroke(128);
-    line(150, 25, 270, 350);  
 
-    stroke(255);
-    line(50, 125, 70, 50);  
+    // set image locations
+    image(imgSetting, 0, 0);
+    image(imgAngryMan, fltAngryManX, fltAngryManY);
+    ellipse(fltCircleX, fltCircleY, 50, 50);
+
+    // add speed
+    fltCircleX += fltCircleSpeedX;
+    fltCircleY += fltCircleSpeedY;
+    
+    // add collision detection
+    if (fltCircleX < 0  || fltCircleX > width - 50) {
+      fltCircleSpeedX *= -1;
+    }
+    if (fltCircleY < 0 || fltCircleY > height - 50) {
+      fltCircleSpeedY *= -1;
+    }
+
+    // add speed
+    fltAngryManX += fltAngryManSpeedX;
+    fltAngryManY += fltAngryManSpeedY;
+
+    // add collision detection
+    if (fltAngryManX < 0  || fltAngryManX > width - 200) {
+      fltAngryManSpeedX *= -1;
+    }
+    if (fltAngryManY < 0 || fltAngryManY > height - 300) {
+      fltAngryManSpeedY *= -1;
+    }
```

Comment:
collision detection demonstrated.  non-linear movement (level 4 requirement) not demonstrated.